### PR TITLE
Fix #8 - Travis CI integration

### LINKED ISSUE #8: Add CI
Add Travis to enable tests and deployment

### PR DESCRIPTION
**Pull request by gideonthomas**

### CODE REVIEW

File: scripts/deploy.js [hunk 1]
```diff
@@ -5,36 +5,32 @@ let pathToRepo = require(`path`).resolve(`.`);
 let getStatus = (repo) => repo.getStatus();
 
 let runDeploy = () => {
+  if(process.env.TRAVIS_PULL_REQUEST) {
+    shell.echo(`Skipping deployment for pull request!`);
```
Comment (gvn) [inline scripts/deploy.js:9]: Nice!

File: scripts/deploy.js [hunk 2]
```diff
@@ -5,36 +5,32 @@ let pathToRepo = require(`path`).resolve(`.`);
...
+    return;
+  }
+
   shell.echo(`Running deployment now...`);
 
-  shell.exec(`git checkout master`);
+  shell.exec(`git stash`);
```
Comment (gvn) [inline scripts/deploy.js:15]: Would there be a dirty repo on Travis for some reason?

Comment (gideonthomas) [inline scripts/deploy.js:15]: This is mainly to clean up anything that might be created by npm test or any other scripts we decide to add later on. It's mainly a precaution but we can remove

File: scripts/deploy.js [hunk 3]
```diff
@@ -34,7 +36,7 @@ let runDeploy = () => {
   shell.exec(`git reset`);
   shell.exec(`git add .`);
   shell.exec(`git commit -m 'deploy.js-ified'`);
```
Comment (gvn) [inline scripts/deploy.js]: We should probably change this to `Deployed via Travis`.

File: scripts/deploy.js [hunk 4]
```diff
@@ -5,21 +5,23 @@ let pathToRepo = require(`path`).resolve(`.`);
...
-  shell.exec(`git checkout master`);
-  shell.exec(`git pull origin master`);
+  shell.echo(`Running deployment now...`);
 
-  shell.rm(`-rf`,`node_modules`);
+  shell.exec(`git stash`);
+  shell.exec(`git fetch origin master:master`)
+  shell.exec(`git checkout --orphan gh-pages master`);
 
-  shell.exec(`npm i`);
```
Comment (gvn) [inline scripts/deploy.js:15]: I think we should leave this in, right?

File: scripts/deploy.js [hunk 5]
```diff
@@ -5,21 +5,23 @@ let pathToRepo = require(`path`).resolve(`.`);
...
   shell.exec(`npm run build`);
 
   shell.echo(`${new Date}\n\n\n`).to(`last-built.txt`);
   shell.exec(`git log -n 1 >> last-built.txt`);
 
-  shell.exec(`git branch -D gh-pages`);
-  shell.exec(`git checkout --orphan gh-pages`);
+  shell.mv(`dest/*`, `./`);
```
Comment (gvn) [inline scripts/deploy.js]: This is now in here 2 times

File: scripts/deploy.js (hunk 4 above)
Comment (gideonthomas) [inline scripts/deploy.js:15]: Travis already does this automatically

File: scripts/deploy.js (hunk 5 above)
Comment (gideonthomas) [inline scripts/deploy.js]: oh whoops

**Pull request merged**